ksqldb-serde/src/main/java/io/confluent/ksql/serde/json/KsqlJsonSerdeFactory.java

```diff
@@ -124,14 +133,33 @@ private <T> Serializer<T> createSerializer(
   }
 
   private <T> Deserializer<T> createDeserializer(
+      final KsqlConfig ksqlConfig,
       final Schema schema,
-      final Class<T> targetType
+      final Class<T> targetType,
+      final ConnectDataTranslator dataTranslator,
+      final Converter converter
   ) {
-    return new KsqlJsonDeserializer<>(
-        schema,
-        useSchemaRegistryFormat,
-        targetType
-    );
+    if (useSchemaRegistryFormat
+        && ksqlConfig.getBoolean(KsqlConfig.KSQL_JSON_SR_CONVERTER_DESERIALIZER_ENABLED)) {
+      return new KsqlConnectDeserializer<>(
+          converter,
+          dataTranslator,
+          targetType
+      );
+    } else {
+      if (useSchemaRegistryFormat) {
+        LOG.info("The JsonSchemaConverter for deserialization is disabled for JSON_SR.");
```

Comment:
Remove this log? It's possible this serde will be called many time, this log can spam log. I guess this can be deduced from the value of `KSQL_JSON_SR_CONVERTER_DESERIALIZER_ENABLED` which might be logged as config.

Reply:
I removed it. I was hesitating to add it in the first place. But you're right, we can deduct this by looking at the config.